fews_3di/simulation.py

```diff
@@ -300,36 +302,38 @@ def _add_laterals(self, laterals):
             if not still_to_process:
                 return
 
-    def _add_initial_state(self, saved_state_id_file: Path):
+    def _add_initial_state(self, saved_state_id_file: Path, cold_state_id_file: Path):
         # TODO explain rationale. (likewise for the other methods).
-        if not saved_state_id_file.exists():
-            msg = f"Saved state id file {saved_state_id_file} not found"
-            if self.allow_missing_saved_state:
-                logger.warn(msg)
-                return
-            else:
-                raise utils.MissingFileException(msg)
-        saved_state_id: str = saved_state_id_file.read_text().strip()
-        logger.info("Simulation will use initial state %s", saved_state_id)
-        try:
-            self.simulations_api.simulations_initial_saved_state_create(
-                self.simulation_id, data={"saved_state": saved_state_id}
-            )
-        except openapi_client.exceptions.ApiException as e:
-            if e.status == 400:
-                logger.debug("Saved state setting error: %s", str(e))
-                msg = (
-                    f"Setting initial state to saved state id={saved_state_id} failed. "
-                    f"The error response was {e.body}, perhaps use "
-                    f"--allow-missing-saved-state initially?"
-                )
+        for state_file in [saved_state_id_file,cold_state_id_file]:
+            if not state_file.exists():
+                msg = f"Saved state id file {state_file} not found"
                 if self.allow_missing_saved_state:
                     logger.warn(msg)
                     return
                 else:
-                    raise MissingSavedStateError(msg) from e
-            logger.debug("Error isn't a 400, so we re-raise it.")
-            raise
+                    raise utils.MissingFileException(msg)
+            saved_state_id: str = state_file.read_text().strip()
+            logger.info("Simulation will use initial state %s", saved_state_id)
+            try:
+                self.simulations_api.simulations_initial_saved_state_create(
+                    self.simulation_id, data={"saved_state": saved_state_id}
+                )
+                return
+            except openapi_client.exceptions.ApiException as e:
+                if e.status == 400:
+                    logger.debug("Saved state setting error: %s", str(e))
+                    msg = (
+                        f"Setting initial state to saved state id={saved_state_id} failed. "
+                        f"The error response was {e.body}, perhaps use "
+                        f"--allow-missing-saved-state initially?"
+                    )
+                    if self.allow_missing_saved_state:
+                        logger.warn(msg)
+                        return
+                else:
+                    ogger.debug("Error isn't a 400, so we re-raise it.")
+                    raise
+                    continue
```

Comment:
zowel raise als continue kan niet kloppen

Reply:
Ik ben in de war. Ik wil dat hij een error raised als er zowel geen cold als geen warm state is.

Reply:
De raise raist prima, maar dan komt 'ie "dus" niet op de volgende regel terecht.